src/python/pants/backend/python/target_types.py

```diff
@@ -486,6 +486,29 @@ class PexIgnoreErrorsField(BoolField):
     help = "Should PEX ignore errors when it cannot resolve dependencies?"
 
 
+class PexShBootField(BoolField):
+    alias = "sh_boot"
+    default = False
```

Comment:
Since people very likely want to set this to True, should we consider a deprecation cycle to change this to be the default?   In a clean world with no backwards compatibility considerations I imagine this would have been the default in Pex.

And/or, should there be an option to set this globally?

Reply:
Will merge now, since this can be done in a followup

Reply:
> Since people very likely want to set this to True, should we consider a deprecation cycle to change this to be the default?

I *think* Pex tries to support constrained environments without `sh`, so I'm not sure of the hypothetical default there.  But yeah for Pants I think this is  value that is more likely to "just work".

> And/or, should there be an option to set this globally?

Isn't `__defaults__` or macros the usual answer in that case?

Reply:
Yeah, actually, we should just document the use of `__defaults__` for this.